message_parser/src/main/java/cumulus_message_adapter/message_parser/MessageAdapter.java

```diff
@@ -47,6 +50,16 @@ public String CallMessageAdapterFunction(String messageAdapterFunction, String i
             writer.write(inputJson);
             writer.close();
 
+            // Honestly i'm not sure why scanner doens't work below, i think the
+            // issue is taht this needs to be consumed before the process will
+            // exit (too large for processbuilder?)
+            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+            String line;
+            //If we ever return more than a single (even very large line) from cumulus_message_adapter this won't work.
+            while ((line = reader.readLine()) != null){
```

Comment:
I'm fine with this. Not that I don't have feelings about it, but you're not fundamentally changing the way any of it works, so I'm not going to comment on that here. 🙂

Reply:
I too have opinions on this. The way the CMA is setup, however, i felt adding the "just in case" was out of scope and don't have a valid test case for it to do integrations with.

Reply:
Completely agreed!